githubapp/client_creator.go

```diff
@@ -201,7 +202,11 @@ func (c *clientCreator) newClient(base *http.Client, details string) (*github.Cl
 		return nil, errors.Wrapf(err, "failed to parse base URL: %q", c.v3BaseURL)
 	}
 
-	client := github.NewClient(base)
+	cache := httpcache.NewMemoryCache()
+	cachedTransport := httpcache.NewTransport(cache)
+	cachedTransport.Transport = base.Transport
+
+	client := github.NewClient(cachedTransport.Client())
```

Comment:
Maybe this won't work, but I was thinking this would be optional middleware in the same way that client logging and metrics are. Since the cache transport is an implementation of `http.RoundTripper` it should be easy to integrate as middleware.

That would also allow users to configure the cache, which I think is a requirement.